### PR TITLE
Update React Native minimum version supported, fix test app autolinking

### PR DESCRIPTION
**Pull request by Saadnajmi**
# Overview

In #227, I accidentally made the minimum react-native dependency `0.73`. Since the PR itself has no source code changes (only test app changes) that wasn't necessary. This PR brings the minimum back down to RN 0.57. I did this by:

- Updating the rnx-kit configs to set development (and test app) to 0.73, while production to 0.57
- Run `yarn rnx-align-deps --write`
- Remove the rnx-kit configs altogether as they aren't actually necessary (more a tool to help with upgrades, and can be used without having a config in your package.json). 
  - This does mean if someone came and ran `npx @rnx-kit/align-deps@latest --requirements react-native@0.74` or something later, it would bump the minimum to 0.74 and not take into account the minimum. I figure nobody will do this, so safe to remove.
  
 Also, from #230, I realized we were no longer auto linking RNCClipboard in our test app. This is due to autolinking not following yarn workspaces. We can fix this by adding the extra `dependencies` key to our `react-native.config.js` to help the react native CLI auto link the library. 
 
 Furthermore, after talking with @tido64 , I realized I had set up the test app a bit off. I had two package.json files: one at the root and one for the example app. With this approach, we would need two react native configs. A more common setup amongst RN community modules is to only have one package.json and react-native config. This meant some more refactoring to bring the test app in line with other repos, and ensure autolinking still worked with Yarn 4.
 
 End result is a working modern test app :)

# Test Plan

Test app boots and loads the module correctly on iOS.

Screenshot 2024-04-10 at 5 05 39 AM

### CODE REVIEW

File: react-native.config.js [hunk 1]
```diff
@@ -1,47 +1,48 @@
...
-        sourceDir: './example/windows',
-        solutionFile: 'ClipboardExample.sln',
-        project: {
-          projectFile: 'ClipboardExample/ClipboardExample.vcxproj',
-        },
+        sourceDir: 'windows',
+        solutionFile: 'windows.sln',
+        projects: [
+          {
+            projectFile: 'ReactNativeWebView/Clipboard.vcxproj',
```
Comment (tido64) [inline react-native.config.js:40]:
```suggestion
            projectFile: 'Clipboard/Clipboard.vcxproj',
```

File: react-native.config.js [hunk 2]
```diff
@@ -1,47 +1,48 @@
...
+        manifestPath: path.join(__dirname, 'example', 'android'),
+      },
+      ios: {
+        sourceDir: 'example/ios',
+      },
+      windows: fs.existsSync('example/windows/Example.sln') && {
+        sourceDir: path.join('example', 'windows'),
+        solutionFile: path.join('example', 'windows', 'Example.sln'),
+        project: path.join(__dirname, 'example', 'windows'),
+      },
```
Comment (tido64) [inline react-native.config.js:19]: There are some unused fields here:
```suggestion
      android: {
        sourceDir: path.join('example', 'android'),
      },
      ios: {
        sourceDir: 'example/ios',
      },
      windows: {
        sourceDir: path.join('example', 'windows'),
        solutionFile: path.join('example', 'windows', 'Example.sln'),
      },
```

Comment (Saadnajmi) [reply react-native.config.js:19]: I got those from web view. Perhaps they're unused there too? I'll check in the branch I have open there.

**Review by Naturalclar — APPROVED** [review]
LGTM!

**Pull request merged**